app/lib/shared/analyzer_client.dart

```diff
@@ -65,4 +86,43 @@ class AnalysisView {
     list.sort();
     return list;
   }
+
+  /// Returns the raw health score between -1.0 and 1.0
+  /// These need to be normalized for search and frontend purposes.
+  double get health {
+    if (_data == null) return 0.0; // missing analysis
+    if (!hasAnalysisData) return -1.0; // aborted analysis
+
+    // Starting out with max.
+    double score = 1.0;
+
+    // Penalty for each major issue.
+    if (_summary.issues != null && _summary.issues.isNotEmpty) {
+      score -= _summary.issues.length * 0.2;
```

Comment:
major issue: `dartfmt`, `dartanalyzer` or `pub` finishing with non-zero exit code or some unexpected error while running their output.

`analyzer_options.yaml` could disable hints/warnings, that is true, but that goes to the minor issues down below. I'm not sure whether we want to override `analyzer_options.yaml`. Moved discussion to https://github.com/dart-lang/pana/issues/63